src/Contracts/MASA.Scheduler.Contracts.Server/Model/BaseServiceModel.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Scheduler.Contracts.Server.Model;
+
+public class BaseServiceModel
+{
+    public Guid ProgramId { get; set; }
+
+    public string HttpHost { get; set; } = string.Empty;
+
+    public string HttpsHost { get; set; } = string.Empty;
+
+    public int HttpPort { get; set; }
+
+    public int HttpsPort { get; set; }
+
+    public DateTimeOffset LastResponseTime { get; set; }
+
+    public ServiceStatuses Status { get; set; }
+
+    public int NotResponseCount { get; set; }
+
+    public string HeartbeatApi { get; set; } = string.Empty;
+
+    public HttpClient CallerClient { get; set; } = default!;
+
+    public string GetServiceUrl(bool containsScheme = true)
+    {
+        if (HttpsPort != 0)
```

Comment:
HttpsPort  has value but HttpsHost no value,HttpsHost  should main condition and handle 80、443